eWonConnector-gateway/src/main/java/com/hms_networks/americas/sc/ignition/data/TagManager.java

```diff
@@ -0,0 +1,870 @@
+              INITIALIZED_EWON_TAGS.remove(ewonName);
+              FORCED_M2WEB_EWONS.remove(ewonName);
+            }
+          }
+
+          // Log tag deletion
+          if (connectorSettings.isDebugEnabled()) {
+            LOGGER.debug(
+                "Tag '{}' on Ewon Flexy '{}' has been deleted from Ignition. "
+                    + "Note: The tag on the Ewon Flexy device is unchanged.",
+                tagName,
+                ewonName);
+          }
+
+          return DeletionHandler.DeletionResponse.Allowed;
+        });
+
+    // Verify integrity of tag history storage information if tag history enabled
+    if (connectorSettings.isHistoryEnabled()
+        && StringUtils.isBlank(connectorSettings.getHistoryProvider())) {
+      LOGGER.warn(
+          "History synchronization is enabled, but no history provider "
+              + "has been specified for storage. No data will be stored");
+    }
+
+    // Output warning if tag name check is disabled
+    if (connectorSettings.isTagNameCheckDisabled()) {
+      LOGGER.warn(
+          "Strict checking for allowed tag name characters is "
+              + "disabled. Tags with unsupported characters may not work properly");
+    } else {
+      LOGGER.warn(
+          "Strict checking for allowed tag name characters is enabled. Tags with unsupported"
+              + " characters will be sanitized and illegal characters replaced with '{}'",
+          TagManagerUtilities.ILLEGAL_TAG_NAME_CHARACTER_REPLACEMENT);
+    }
+  }
+
+  /**
+   * Shuts down the tag manager.
+   *
+   * @since 1.0.0
+   */
+  public static void shutdown() {
+    if (tagProvider != null) {
+      tagProvider.shutdown(false);
+      tagProvider = null;
+    }
+
+    // Clear all lists and maps
+    FORCED_M2WEB_EWONS.clear();
+    INITIALIZED_EWON_TAGS.clear();
+  }
+
+  /**
+   * Gets the list of Ewon Flexy devices that should be polled via M2Web.
+   *
+   * @implNote This list is populated with Ewon Flexy devices which have the 'forceRealtimeData'
+   *     option set to 'true'. If the {@link EwonConnectorSettings#isForceLive()} option is set to
+   *     'true', this method will return the same result, as this option should be handled by the
+   *     calling method.
+   * @return the list of Ewon Flexy devices that should be polled via M2Web
+   * @since 1.0.0
+   */
+  public static List<M2WebEwon> getForcedM2WebEwons() {
+    synchronized (FORCED_M2WEB_EWONS) {
+      return FORCED_M2WEB_EWONS;
+    }
+  }
+
+  /**
+   * Checks if the specified Ewon Flexy device is configured to be polled via M2Web.
+   *
+   * @param ewonName Ewon Flexy name to check
+   * @return {@code true} if the Ewon Flexy device is configured to be polled via M2Web, {@code
+   *     false} otherwise
+   * @since 1.0.0
+   */
+  public static boolean isEwonForcedM2Web(String ewonName) {
+    synchronized (FORCED_M2WEB_EWONS) {
+      return FORCED_M2WEB_EWONS.contains(ewonName);
```

Comment:
Again I see docs for the method Collection.contains(object) method. Not sure how Collection.conatins(string) works.

Reply:
Will correct